plugins/reporters/trustsource/src/main/kotlin/TrustSourceModel.kt

```diff
@@ -22,34 +22,53 @@ package org.ossreviewtoolkit.plugins.reporters.trustsource
 import kotlinx.serialization.Serializable
 
 @Serializable
-data class TrustSourceModule(
+data class NewScan(
+    /** Name of the module */
     val module: String,
-    val moduleId: String,
-
-    val dependencies: List<TrustSourceDependency>
+    /** SCM tag of the module */
+    val tag: String = "",
+    /** SCM branch of the module */
+    val branch: String = "",
+    /** List of the module's dependencies */
+    val dependencies: List<Dependency>
 )
 
 @Serializable
-data class TrustSourceDependency(
-    val key: String,
+data class Dependency(
     val name: String,
-    val repoUrl: String,
-    val homepageUrl: String,
-    val description: String,
-    val checksum: String,
-    val private: Boolean,
-    val versions: List<String>,
-
-    val dependencies: List<TrustSourceDependency>,
-    val licenses: List<TrustSourceLicense>,
-    val meta: TrustSourceMeta
+    val purl: String,
+    val dependencies: List<Dependency>,
+
+    val description: String = "",
+    /** Package homepage, may differ or may be the same with the repositoryUrl **/
+    val homepageUrl: String = "",
+    /** Repository with the package's source code **/
+    val repositoryUrl: String = "",
+
+    /** Map where a key denotes a checksum algorithm and the value is the checksum itself **/
+    val checksum: Map<String, String> = emptyMap(),
+    val licenses: List<License> = emptyList(),
+
+    /** Dependencies source package **/
+    val `package`: Package? = null,
```

Comment:
For ease of referencing the property, we could name it "pkg" but have a `@SerialName("package")` annotation.

Reply:
Done